diff --git a/CMakeLists.txt b/CMakeLists.txt
index d353cc9..ee6a7d1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,7 @@ cmake_minimum_required(VERSION 3.14.0 FATAL_ERROR)
 project(exawind CXX C Fortran)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON CACHE INTERNAL "")
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+include(exawind-utils)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_EXTENSIONS OFF)
@@ -10,6 +11,7 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 option(EXAWIND_TEST_WITH_FCOMPARE "Check AMR-Wind test plots against gold files" OFF)
 option(EXAWIND_SAVE_GOLDS "Provide a directory in which to save golds during testing" OFF)
+option(EXAWIND_ENABLE_CUDA "Enable CUDA" OFF)
 
 set(CMAKE_PREFIX_PATH ${Trilinos_DIR} ${TIOGA_DIR}
                       ${Nalu-Wind_DIR} ${AMR-Wind_DIR}
@@ -25,6 +27,10 @@ find_package(Filesystem REQUIRED)
 if (NALU_USES_OPENFAST)
   find_package(OpenFAST REQUIRED)
 endif()
+if (EXAWIND_ENABLE_CUDA)
+  enable_language(CUDA)
+  find_package(CUDAToolkit REQUIRED)
+endif()
 
 # General information about machine, compiler, and build type
 message(STATUS "Exawind information:")
@@ -46,6 +52,13 @@ set(EXAWIND_EXE_NAME "exawind")
 add_executable(${EXAWIND_EXE_NAME})
 add_subdirectory(app/exawind)
 
+if(EXAWIND_ENABLE_CUDA)
+  set(ewtargets "${EXAWIND_LIB_NAME};${EXAWIND_EXE_NAME}")
+  foreach(tgt IN LISTS ewtargets)
+    set_cuda_build_properties(${tgt})
+  endforeach()
+endif()
+
 enable_testing()
 include(CTest)
 add_subdirectory(test)
diff --git a/app/exawind/CMakeLists.txt b/app/exawind/CMakeLists.txt
index 9bdefeb..d50f95b 100644
--- a/app/exawind/CMakeLists.txt
+++ b/app/exawind/CMakeLists.txt
@@ -1,8 +1,8 @@
 target_sources(${EXAWIND_EXE_NAME} PRIVATE
   exawind.cpp)
 
-target_link_libraries(${EXAWIND_EXE_NAME} PUBLIC
+target_link_libraries(${EXAWIND_EXE_NAME} PRIVATE
   ${EXAWIND_LIB_NAME}
   yaml-cpp)
 
-target_include_directories(${EXAWIND_EXE_NAME} SYSTEM PUBLIC ${YAML_CPP_INCLUDE_DIR})
+target_include_directories(${EXAWIND_EXE_NAME} SYSTEM PRIVATE ${YAML_CPP_INCLUDE_DIR})
diff --git a/cmake/exawind-utils.cmake b/cmake/exawind-utils.cmake
new file mode 100644
index 0000000..055656c
--- /dev/null
+++ b/cmake/exawind-utils.cmake
@@ -0,0 +1,9 @@
+function(set_cuda_build_properties target)
+  get_target_property(_tgt_src ${target} SOURCES)
+  list(FILTER _tgt_src INCLUDE REGEX "\\.cpp")
+  set_source_files_properties(${_tgt_src} PROPERTIES LANGUAGE CUDA)
+  set_target_properties(${target} PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
+  set_target_properties(${target} PROPERTIES CUDA_ARCHITECTURES "70")
+  set_target_properties(${target} PROPERTIES CUDA_RESOLVE_DEVICE_SYMBOLS ON)
+  #set_target_properties(${target} PROPERTIES LINKER_LANGUAGE CUDA)
+endfunction(set_cuda_build_properties)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e872dac..abbd720 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -19,5 +19,7 @@ target_include_directories(${EXAWIND_LIB_NAME} PUBLIC
 
 target_link_libraries(${EXAWIND_LIB_NAME} PUBLIC
   AMR-Wind::amrwind_api
+  AMR-Wind::buildInfoamrwind_obj
+  AMReX::amrex
   Nalu-Wind::nalu
   std::filesystem)
