diff --git a/packages/stk/stk_mesh/stk_mesh/base/NgpFieldSyncDebugger.hpp b/packages/stk/stk_mesh/stk_mesh/base/NgpFieldSyncDebugger.hpp
index 8001e363ddc..e85727613a8 100644
--- a/packages/stk/stk_mesh/stk_mesh/base/NgpFieldSyncDebugger.hpp
+++ b/packages/stk/stk_mesh/stk_mesh/base/NgpFieldSyncDebugger.hpp
@@ -58,7 +58,7 @@ class EmptyNgpFieldSyncDebugger
 public:
   using StkFieldSyncDebuggerType = EmptyStkFieldSyncDebugger;
 
-  KOKKOS_DEFAULTED_FUNCTION EmptyNgpFieldSyncDebugger(StkFieldSyncDebuggerType *) {}
+  KOKKOS_FUNCTION EmptyNgpFieldSyncDebugger(StkFieldSyncDebuggerType *) {}
   KOKKOS_DEFAULTED_FUNCTION ~EmptyNgpFieldSyncDebugger() = default;
 
   KOKKOS_DEFAULTED_FUNCTION EmptyNgpFieldSyncDebugger(const EmptyNgpFieldSyncDebugger &) = default;
