diff --git a/tools/build/src/engine/build.sh b/tools/build/src/engine/build.sh
index f1ad08cb..fea22e12 100755
--- a/tools/build/src/engine/build.sh
+++ b/tools/build/src/engine/build.sh
@@ -329,8 +329,8 @@ case "${B2_TOOLSET}" in
 
     intel-*)
         CXX_VERSION_OPT=${CXX_VERSION_OPT:---version}
-        B2_CXXFLAGS_RELEASE="-O3 -s -static"
-        B2_CXXFLAGS_DEBUG="-O0 -g -p -static"
+        B2_CXXFLAGS_RELEASE="-O3 -s"
+        B2_CXXFLAGS_DEBUG="-O0 -g -p"
     ;;
 
     vacpp)
